### ISSUE TITLE
🐛 Bug: SingleSelect options do not keep their order

### ISSUE THREAD

**Issue by djmaze** [nocodb/nocodb#4429]
### Please confirm if bug report does NOT exists already ?

- [X] I confirm there is no existing issue for this

### Steps to reproduce ?

1. Create a new table
2. Add a SingleSelect field with multiple options (e.g. `a`, `b` and `c`)
3. Add a new record and look at the order of the options in the select field
4. Open the field definition editor and look at the order of the options.
5. Reorder the options and save the definition.
6. Edit the field definition again and look at the order of the options.

### Desired Behavior

The select options should always be shown in the order of the field definition. That means in step 3,4 and 6, the order should be `a`, `b`, `c` (unless I have reordered them in the field definition editor).

### Project Details

NocoDB used as docker: true
NocoDB version: 0.98.4
Database used in NC_DC URL: pg
Project was created by clicking: New Project
Database on which spreadsheet is created: pg
OS on which NocoDB is running: Docker host is Ubuntu 22.04
Database version: Postgres 15.0

### Attachements

(Options were initially entered in the following order: `a`, `b`, `c`)

![Definition_order](https://user-images.githubusercontent.com/7229/202748070-2d7ef17f-a777-48a1-b6a6-4f02fa6d1602.png)
![Edit_order](https://user-images.githubusercontent.com/7229/202748093-62c75813-e092-4212-b6b4-cd64a27b9e29.png)

**Comment by wingkwong** [comment]
Are you able to reproduce the issue in a new column? I try to follow the steps and it's working as expected.

**Comment by djmaze** [comment]
Yep, the problem persists.

Maybe it's related to my postgres version (15)?

**Comment by djmaze** [comment]
Looking at the developer console, when loading the page I can see a xhr request to `/md_xxx` whose result contains a `columns` property. Inside that I can find an entry with a `title` set to `Status` (the field name) and the field `dtxp` set to "'a','b','c'"`.

So it seems to me the backend returns the values in the correct order. In the frontend though, the order is mixed up in the column definition editor as well as in the value edit field in the table.

**Comment by djmaze** [comment]
I had another close look and there is another value in the backend response (`colOptions`) which returns the value in reverse order, but with the `order` value correctly set:

![backend_values](https://user-images.githubusercontent.com/7229/203439834-3ae9d86d-2c77-4fae-953d-eac0086411a5.png)

So the backend seemed the culprit again.

As I could not reproduce this bug in a development setup of NocoDB (it worked correctly there), I thought it might be related to my production setup. So I remembered I am using _Redis_ there.

It turns out after restarting Redis, the order was correct again. Whats more, after each modification to the option order, I have to restart Redis in order to fix the display order.

After removing Redis from the configuration, everything seems to be working fine permanently.

**TL/DR:** It looks to me like Redis is caching the column metadata incorrectly. I could also reproduce this problem on the latest master branch.

**Comment by wingkwong** [comment]
Thanks for the findings. Will check later.

**Comment by djmaze** [comment]
Thanks!
